lib/extract-templates.js

```diff
@@ -0,0 +1,164 @@
+import { parseTemplates } from 'ember-template-imports/lib/parse-templates.js';
+import * as util from 'ember-template-imports/src/util.js';
+
+export const SUPPORTED_EXTENSIONS = ['.js', '.ts', '.gjs', '.gts'];
+const LOCATION_START = Object.freeze({ line: 0, column: 0 });
+/**
+ * @typedef {object} TemplateInfo
+ * @property {number} line
+ * @property {number} column
+ * @property {string} template
+ * @property {boolean} isEmbedded
+ *
+ * @param {string} moduleSource
+ * @param {string} relativePath
+ *
+ * @returns {TemplateInfo[]}
+ */
+export function extractTemplates(moduleSource, relativePath) {
+  // If no relativePath is present, assuming we might have templates.
+  let mightHaveTemplates = relativePath ? isSupportedScriptFileExtension(relativePath) : true;
+
+  if (!mightHaveTemplates) {
+    return [makeTemplateInfo(LOCATION_START, moduleSource)];
+  }
+
+  let parsed = parseTemplates(moduleSource, relativePath, {
+    templateTag: util.TEMPLATE_TAG_NAME,
+    templateLiteral: [
+      {
+        importPath: 'ember-cli-htmlbars',
+        importIdentifier: 'hbs',
+      },
+      {
+        importPath: '@ember/template-compilation',
+        importIdentifier: 'hbs',
+      },
+      {
+        importPath: util.TEMPLATE_LITERAL_MODULE_SPECIFIER,
+        importIdentifier: util.TEMPLATE_LITERAL_IDENTIFIER,
+      },
+      {
+        importPath: 'ember-cli-htmlbars-inline-precompile',
+        importIdentifier: 'default',
+      },
+      {
+        importPath: 'htmlbars-inline-precompile',
+        importIdentifier: 'default',
+      },
+      {
+        importPath: '@ember/template-compilation',
+        importIdentifier: 'precompileTemplate',
+      },
+    ],
+  });
```

Comment:
In order to go this route, I think we need to deal with https://github.com/ember-template-imports/ember-template-imports/issues/42. We need to know (from the return value of `parseTemplates`) _which_ of these imports were used (so that we know things like "is this strict mode or loose mode?").

We might be able to start off without that info, but I think it will eventually be required. The experience (in editor) when using `<template>` should tell you when you are accidentally relying on `<Input />` or `{{hash}}`/`{{get}}` without an import ([see the RFC](https://github.com/emberjs/rfcs/blob/master/text/0496-handlebars-strict-mode.md#keywords) for details on keywords vs helpers).

Reply:
What I have implemented so far is kind of a "best effort" approach, because we can't always know the file type based on the information passed to `ember-template-lint` (see the [discord conversation about this](https://discord.com/channels/480462759797063690/518154533143183377/967169732849598474)).

I'm down to either wait for https://github.com/ember-template-imports/ember-template-imports/issues/42 to be done (looks like @nlfurniss already made some progress) or help complete it, but what I've done here is the best we can do with the information we have so far.

Let's resolve the conversation about this here.